x-pack/plugins/maps/public/layers/styles/abstract_style.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ReactElement } from 'react';
+import { StyleDescriptor } from '../../../common/descriptor_types';
+import { ILayer } from '../layer';
+
+// todo: not sure if this is the right one
+export interface IStyle {
+  getDescriptor(): StyleDescriptor;
+  getDescriptorWithMissingStylePropsRemoved(): unknown;
```

Comment:
Parameters are `nextFields: IField[]`
Return type is `{ hasChanges: boolean, nextStyleDescriptor: StyleDescriptor }`